### PR TITLE
Addressed Zone heating load reporting issue if Zone Heating Design Supply Air Temperature is less than the heating setpoint

### LINKED ISSUE #7039: Zone sizing reports zero heating load if Zone Heating Design Supply Air Temperature is less than the heating setpoint
Issue overview
--------------
This situation should produce non-zero design heating loads, issue a warning, and report zero design heating airflow rates (because the air flow is undefined for this situation). Look at the classroom zones, for example, CORNER_CLASS_1_POD_1_ZN_1_FLR_1 ZN, in the HVAC Sizing Summary, Zone Sensible Heating table.

### Workaround
Set the `Sizing:Zone` Zone Heating Design Supply Air Temperature to be greater than the thermostat heating setpoint.

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 9.0.1-bb7ca4f0da
 - Unmethours [35186](https://unmethours.com/question/35186/zone-design-loads-do-not-account-for-doas/)
### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file attached here  [in-MJW-FixShadowCalc-v9.0.idf.txt](https://github.com/NREL/EnergyPlus/files/2527379/in-MJW-FixShadowCalc-v9.0.idf.txt)

 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
Fixes #7039. This fix allows reporting zone heating/cooling loads when the zone design flow rate is set to zero due to the Design Supply Air Temperature is less than the zone air temperature at peak heating load for design heating flow calculation or Design Supply Air Temperature is greater than the the zone air temperature at peak cooling load for design cooling flow calculation.  Now, the users will see the design cooling and heating loads even if the design flow rates are set zero and appropriate warning error message will also be displayed. Diffs are expected/justified based on this change if the design specification and zone condition combination allows.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by Nigusse** [comment]
@mjwitte  This fix checks the calculated design heating/cooling load and reports the calculated values even if the design flow rates are zero. Now the calculated load sequence of the sizing period are also reported. Also throws error message if the design flow rates were set to zero if design supply air temperature is less than the zone air temperature at peak heating load, vice versa for cooling case.  Below is sample error message from the defect file: 

** Severe  ** UpdateZoneSizing: Supply air temperature is less than zone temperature during heating air flow calculations
   **   ~~~   ** ...calculated design heating volume flow rate = 0.00000 m3/s
   **   ~~~   ** ...calculated design heating mass flow rate   = 0.00000 kg/s
   **   ~~~   ** ...thermostat set piont temp   = 21.111 C
   **   ~~~   ** ...zone temperature            = 21.111 C
   **   ~~~   ** ...supply air temperature      = 12.778 C
   **   ~~~   ** ...occurs in zone              = CORNER_CLASS_1_POD_1_ZN_1_FLR_1 ZN
   **   ~~~   ** ...Note: supply air temperature should be greater than zone temperature during heating air flow calculations

**Comment by Nigusse** [comment]
Review of regression results shows that some thing is broken that was not intended. Investigating the code changes.

**Comment by Myoldmopar** [comment]
CI results are good now, one Mac failure is unrelated as well as the Mac build warnings.  However, this branch is conflicted.  I will try to do a quick conflict resolution, but if it gets too in-depth I'll send this back for a resolution.

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 1]
```diff
@@ -2656,7 +2716,7 @@ namespace ZoneEquipmentManager {
                                 }
                                 ShowContinueError("...check zone thermostat set point and design supply air temperatures");
                                 ShowContinueError("...zone name = " + CalcFinalZoneSizing(I).ZoneName);
-                                ShowContinueError("...design heating load         = " + RoundSigDigits(CalcFinalZoneSizing(I).DesCoolLoad, 2) + " W");
+                                ShowContinueError("...design heating load         = " + RoundSigDigits(CalcFinalZoneSizing(I).DesHeatLoad, 2) + " W");
                                 ShowContinueError("...thermostat set point temp   = " + RoundSigDigits(CalcFinalZoneSizing(I).HeatTstatTemp, 3) +
```
Comment (Myoldmopar) [inline src/EnergyPlus/ZoneEquipmentManager.cc:2720]: The merge conflict was just that you had changed DesCoolLoad to DesHeatLoad, and a different branch had fixed a typo from piont to point.  Easy enough.  I'll let CI have another pass over this updated version.

**Comment by Myoldmopar** [comment]
My merge using Github's online conflict resolver left a unit test issue in place.  I fixed it and pushed that up.

File: tst/EnergyPlus/unit/ReportSizingManager.unit.cc [hunk 2]
```diff
@@ -928,3 +925,504 @@ TEST_F(EnergyPlusFixture, ReportSizingManager_FanPeak)
...
+        "      120.;                    !- Field 4",
+
+        "    ScheduleTypeLimits,",
+        "      Any Number;              !- Name",
+
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    SimulationManager::ManageSimulation();
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ReportSizingManager.unit.cc:1410]: Hopefully we'll be able to set something up where we don't have to call the entire ManageSimulation function, but I understand that it works well for now.

File: tst/EnergyPlus/unit/ReportSizingManager.unit.cc [hunk 3]
```diff
@@ -928,3 +925,504 @@ TEST_F(EnergyPlusFixture, ReportSizingManager_FanPeak)
...
+    EXPECT_EQ(DataSizing::CalcFinalZoneSizing(CtrlZoneNum).HeatDesTemp, 12.0);   // less than zone air Temp
+    EXPECT_EQ(DataSizing::CalcFinalZoneSizing(CtrlZoneNum).HeatDesDay, "PHOENIX SKY HARBOR INTL AP ANN HTG 99.6% CONDNS DB");
+    // actual zone air temperature at peak load
+    EXPECT_NEAR(DataSizing::CalcFinalZoneSizing(CtrlZoneNum).ZoneTempAtHeatPeak, 17.07, 0.01);
+    EXPECT_NEAR(DataSizing::FinalZoneSizing(CtrlZoneNum).ZoneTempAtHeatPeak, 17.07, 0.01);
+    // Check heating design flow rates, expected to be zero due the above conditions
+    EXPECT_EQ(DataSizing::CalcFinalZoneSizing(CtrlZoneNum).DesHeatVolFlow, 0.0);  // expects zero
+    EXPECT_EQ(DataSizing::CalcFinalZoneSizing(CtrlZoneNum).DesHeatMassFlow, 0.0); // expects zero
+    EXPECT_EQ(DataSizing::FinalZoneSizing(CtrlZoneNum).DesHeatVolFlow, 0.0);      // expects zero
+    EXPECT_EQ(DataSizing::FinalZoneSizing(CtrlZoneNum).DesHeatMassFlow, 0.0);     // expects zero
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ReportSizingManager.unit.cc:1424]: Makes sense

File: tst/EnergyPlus/unit/ReportSizingManager.unit.cc [hunk 4]
```diff
@@ -928,3 +925,504 @@ TEST_F(EnergyPlusFixture, ReportSizingManager_FanPeak)
...
+    // expects non-zero peak heating load
+    EXPECT_NEAR(DataSizing::CalcFinalZoneSizing(CtrlZoneNum).DesHeatLoad, 6947.94, 0.01);
+    EXPECT_NEAR(DataSizing::FinalZoneSizing(CtrlZoneNum).DesHeatLoad, 6947.94, 0.01);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ReportSizingManager.unit.cc]: With thermostat temp of 21, we'd still expect this heating load.  OK.

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 5]
```diff
@@ -2673,10 +2733,18 @@ namespace ZoneEquipmentManager {
...
+                                ShowContinueError("...calculated design heating mass flow rate   = " +
+                                                  RoundSigDigits((CalcFinalZoneSizing(I).DesHeatMassFlow), 5) + " kg/s");
+                                ShowContinueError("...thermostat set piont temp   = " + RoundSigDigits(CalcFinalZoneSizing(I).HeatTstatTemp, 3) +
+                                                  " C");
                                 ShowContinueError("...zone temperature            = " + RoundSigDigits(CalcFinalZoneSizing(I).ZoneTempAtHeatPeak, 3) +
                                                   " C");
                                 ShowContinueError("...supply air temperature      = " + RoundSigDigits(SupplyTemp, 3) + " C");
                                 ShowContinueError("...occurs in zone              = " + CalcFinalZoneSizing(I).ZoneName);
+                                ShowContinueError("...Note: supply air temperature should be greater than zone temperature during heating air "
+                                                  "flow calculations");
```
Comment (Myoldmopar) [inline src/EnergyPlus/ZoneEquipmentManager.cc:2747]: :+1:

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 6]
```diff
@@ -2631,10 +2683,18 @@ namespace ZoneEquipmentManager {
                             } else if (std::abs(DeltaTemp) > SmallTempDiff && SupplyTemp > CalcFinalZoneSizing(I).ZoneTempAtCoolPeak) {
                                 ShowSevereError(
                                     "UpdateZoneSizing: Supply air temperature is greater than zone temperature during cooling air flow calculations");
+                                ShowContinueError("...calculated volume flow rate  = " + RoundSigDigits((CalcFinalZoneSizing(I).DesCoolVolFlow), 5) +
+                                                  " m3/s");
+                                ShowContinueError("...calculated mass flow rate    = " + RoundSigDigits((CalcFinalZoneSizing(I).DesCoolMassFlow), 5) +
+                                                  " kg/s");
+                                ShowContinueError("...thermostat set point temp    = " + RoundSigDigits(CalcFinalZoneSizing(I).CoolTstatTemp, 3) +
+                                                  " C");
```
Comment (Myoldmopar) [inline src/EnergyPlus/ZoneEquipmentManager.cc:2691]: :+1:

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Code changes look good.  CI is happy except Windows which ran much later than the others and resulted in unrelated diffs.  I'm running this locally but expect to merge it in soon.

**Comment by Myoldmopar** [comment]
I ran locally, confirmed the zero load sizing values with the develop branch and the properly reported values with this branch.

However, when running the unit test suite on this branch (with develop merged in), the new unit test failed because the zone sizing values had deviated somewhat.

I'm OK with this going in, because the change must've been due to changes from a separate PR.  **However**, @Nigusse, I'd still like you to "approve" my fixes to the unit test values (9da6a848867).  If you specifically expected those exact values and something else broke it, please fix it up.  If not, it's probably OK to go in.

**Comment by Nigusse** [comment]
Let me check and will let you know.

**Comment by Myoldmopar** [comment]
Thanks for taking a look @Nigusse, I'm going to go ahead and merge this in.  

The Mac CI already tried to pick up the branch for testing, and the rest of the CI machines might also try to do so.  So I'll leave the branch alive for right now so we don't end up with a bunch of unnecessary red flags due to the deleted branch.

**Pull request merged**